src/main/java/seedu/jarvis/model/financetracker/PurchaseList.java

```diff
@@ -51,8 +53,12 @@ public void resetData(PurchaseList purchaseList) {
     //=========== Getter Methods ==================================================================================
 
     public Purchase getPurchase(int purchaseIndex) {
-        Index index = Index.fromOneBased(purchaseIndex);
-        return allPurchases.get(index.getZeroBased());
+        try {
```

Comment:
I think the `try` / `catch` statement is not needed here, a guard conditional might be better.

Reply:
why would this be so?